benchmarks/burgers/burgers_driver.cpp

```diff
@@ -123,6 +123,8 @@ TaskCollection BurgersDriver::MakeTaskCollection(BlockList_t &blocks, const int
     // estimate next time step
     if (stage == integrator->nstages) {
       auto new_dt = tl.AddTask(update, EstimateTimestep<MeshData<Real>>, mc1.get());
+      auto tag_refine =
```

Comment:
I think we do need `pmesh->adaptive`. 

Regarding the time step control... I agree in principle that could be a problem but we never encountered any issues, so I wonder if there's something I'm missing there.

Reply:
> I think we do need `pmesh->adaptive`.

Agreed. Its fixed now and I also moved to the `ApplyBoundaryConditionsMD` to keep everything all in the same task region to tag after applying the BC.

> Regarding the time step control... I agree in principle that could be a problem but we never encountered any issues, so I wonder if there's something I'm missing there.

I think because the refined variable is 10x all the other components it dominates the CFL condition. As a little experiment I offset the spatial profile of the non-refined components and gave one of them a 50x multiplier to try and force where the timestep is overestimated before a refinement. There is definitely some non-monotonicity compared to uniformly refined case.
![image](https://github.com/user-attachments/assets/0c917703-b37a-4a2f-bc17-1e0b82fa66ca)
![image](https://github.com/user-attachments/assets/e30f56a6-62a9-4994-9c64-79f525d17bae)